sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitionsSuite.scala

```diff
@@ -118,6 +118,28 @@ class PruneFileSourcePartitionsSuite extends PrunePartitionSuiteBase with Shared
     }
   }
 
+  test("SPARK-38357 data + partition filters with OR") {
```

Comment:
If we don't have this test coverage in `master` branch, please land this test case at `master` branch first with another PR, @huaxingao .